modules/claims_api/app/controllers/claims_api/v1/forms/disability_compensation_controller.rb

```diff
@@ -10,23 +10,26 @@ class DisabilityCompensationController < BaseFormController
         FORM_NUMBER = '526'
         before_action { permit_scopes %w[claim.write] }
         before_action :verification_itf_expiration, only: %i[submit_form_526]
-        before_action :validate_526_payload, only: %i[submit_form_526]
         skip_before_action :validate_json_schema, only: %i[upload_supporting_documents]
         skip_before_action :verify_mvi, only: %i[submit_form_526 validate_form_526]
         skip_before_action :log_request, only: %i[validate_form_526]
 
         def submit_form_526
+          headers = auth_headers
+          service = EVSS::DisabilityCompensationForm::ServiceAllClaim.new(headers)
           auto_claim = ClaimsApi::AutoEstablishedClaim.create(
             status: ClaimsApi::AutoEstablishedClaim::PENDING,
-            auth_headers: auth_headers,
+            auth_headers: headers,
             form_data: form_attributes
           )
           auto_claim = ClaimsApi::AutoEstablishedClaim.find_by(md5: auto_claim.md5) unless auto_claim.id
-          auto_claim.form.to_internal
+          service.validate_form526(auto_claim.form.to_internal)
```

Comment:
Whats the purpose of moving this outside of the before filter?

Reply:
the context of the payload matters, in the case of the create the object is stored but in the case of the validate we don't want to store the object so we run a .new instead, having this in the before filter forces all of the autoclaim code to run twice which didn't make sense so instead I removed the code from the filter (which wasn't being used for the validate end point anyhow) and simplified it here here.

Reply:
Definitely seems like a place we can do a refactor but I suppose this is fine for now. Going to tag the refactor issue to revisit: https://github.com/department-of-veterans-affairs/vets-contrib/issues/2937

Reply:
Thanks for segmenting the refactor into a different ticket.